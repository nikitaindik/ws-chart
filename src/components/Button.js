import React from 'react';
import cx from 'classnames';

import style from './Button.module.css';

const Button = ({ onClick, children, isActive }) => {
  const className = cx(style.button, { [style.active]: isActive });
  return (
    <button className={className} onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
