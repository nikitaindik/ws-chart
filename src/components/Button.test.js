import React from 'react';
import { render } from 'react-testing-library';

import Button from './Button';

test('Button snapshot', () => {
  const mockedProps = {
    onClick: jest.fn(),
  };

  const { container } = render(<Button {...mockedProps}>Hello!</Button>);
  expect(container).toMatchSnapshot();
});
