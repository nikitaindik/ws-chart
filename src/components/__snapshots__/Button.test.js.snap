// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<div>
  <button
    class="button"
  >
    Hello!
  </button>
</div>
`;
